### PR TITLE
feat: initial infrastructure configs

### PR DESCRIPTION
**Pull request by iamnewton**
Co-authored-by: chammon <chammon@rubiconproject.com>
Co-authored-by: Newton Koumantzelis <mail@iamnewton.com>

### CODE REVIEW

**Review by brodrigu — APPROVED** [review]
☕

File: .circleci/config.yml [hunk 1]
```diff
@@ -117,29 +85,21 @@ workflows:
           name: << pipeline.parameters.job-name-deploy-staging >>
           stack_name: << pipeline.parameters.project-name >>-staging
           template: << pipeline.parameters.cluster-template >>
-          args: --capabilities CAPABILITY_NAMED_IAM --no-fail-on-empty-changeset --parameter-overrides Certificate=<< pipeline.parameters.certificate-staging >> Image=<< pipeline.parameters.repo >>/sharedid:<< pipeline.git.revision >> Environment=qa
+          args: --capabilities CAPABILITY_NAMED_IAM --no-fail-on-empty-changeset --parameter-overrides Certificate=<< pipeline.parameters.certificate-staging >> Image=<< pipeline.parameters.repo >>/sharedid:<< pipeline.git.revision >> Environment=qa CPU=256 Memory=512
```
Comment (iamnewton) [inline .circleci/config.yml:82]: Is the CPU/Memory going to change per environment?

File: .circleci/config.yml [hunk 2]
```diff
@@ -63,34 +58,7 @@ executors:
...
-          command: 'package'
-      - maven/test:
-          name: 'Building Docker image'
-          command: 'docker:build'
-      - maven/test:
-          name: 'Pushing Docker image'
-          command: 'docker:push'
-
   publish:
     docker:
```
Comment (iamnewton) [inline .circleci/config.yml:57]:
```suggestion
    executor: docker-executor
```

The executor is global here, so you don’t have to redefine it anymore.

File: .circleci/config.yml [hunk 3]
```diff
@@ -63,34 +58,7 @@ executors:
...
       - image: cimg/openjdk:11.0
```
Comment (iamnewton) [inline .circleci/config.yml:58]: Based on the last comment you can remove this

**Review by iamnewton — COMMENTED** [review]
General comment: These [lines](https://github.com/prebid/shared-id/pull/11/files#diff-78a8a19706dbd2a4425dd72bdab0502ed7a2cef16365ab7030a5a0588927bf47R65-R72) were removed in favor of bash commands.  Was there any reason, so I can understand?

File: .circleci/config.yml (hunk 1 above)
Comment (coryhammon1) [reply .circleci/config.yml:82]: Yes, production boxes are bigger (2048 CPU and 4G memory).

Comment (iamnewton) [reply .circleci/config.yml:82]: Oh, cool.  That makes sense.

**Comment by coryhammon1** [comment]
> General comment: These [lines](https://github.com/prebid/shared-id/pull/11/files#diff-78a8a19706dbd2a4425dd72bdab0502ed7a2cef16365ab7030a5a0588927bf47R65-R72) were removed in favor of bash commands. Was there any reason, so I can understand?

The project uses a maven docker plugin to build the docker image and push to the registry. The standard maven orb doesn't support custom commands for using maven plugins, so the only way I could get it to work was to install maven directly and use bash commands.

**Comment by iamnewton** [comment]
Ahh, I know you can do custom commands, but I think the environment variables that you’re exposing might cause the problem.  Was that what you were seeing?

**Comment by coryhammon1** [comment]
I really just couldn't get the maven orb to cooperate. You can do custom commands, but it wasn't registering the ones I wanted to do for some reason (said they didn't exist).

**Pull request merged**